src/Jackalope/Transport/DoctrineDBAL/Client.php

```
@@ -2467,4 +2491,71 @@ private function initConnection()
 
         $this->connectionInitialized = true;
     }
+
+    /**
+     * Check-in item at path.
+     *
+     * @see VersionManager::checkin
+     *
+     * @param string $path absolute path to the node
+     *
+     * @return string path to the new version
+     *
+     * @throws UnsupportedRepositoryOperationException
+     * @throws RepositoryException
+     */
+    public function checkinItem($path)
+    {
+        return $this->getVersionHandler()->checkinItem($path);
+    }
+
+    /**
+     * Check-out item at path.
+     *
+     * @see VersionManager::checkout
+     *
+     * @param string $path absolute path to the node
+     *
+     * @throws UnsupportedRepositoryOperationException
+     * @throws RepositoryException
+     */
+    public function checkoutItem($path)
+    {
+        return $this->getVersionHandler()->checkoutItem($path);
+    }
+
+    /**
+     * Restore the item at versionPath to the location path
+     *
+     * TODO: This is incomplete. Needs batch processing to avoid chicken-and-egg problems
+     *
+     * @see VersionManager::restoreItem
+     */
+    public function restoreItem($removeExisting, $versionPath, $path)
+    {
+        // TODO: Implement restoreItem() method.
```

please throw a NotImplementedExeption rather than silently failing.